pages/blog/[slug].js

```diff
@@ -0,0 +1,54 @@
+import { createClient } from "contentful";
+import Image from "next/image";
+import { documentToReactComponents } from "@contentful/rich-text-react-renderer";
+
+const client = createClient({
+  space: process.env.CONTENTFUL_SPACE_ID,
+  accessToken: process.env.CONTENTFUL_ACCESS_TOKEN,
+});
+
+export const getStaticPaths = async () => {
+  const res = await client.getEntries({
+    content_type: "blog",
+  });
```

Comment:
This particular code block is being repeated across the app. Could this be a function instead?